libarchive/archive_write_set_format_xar.c

```diff
@@ -3252,4 +3290,120 @@ getalgname(enum sumalg sumalg)
 	}
 }
 
+#if HAVE_LIBXML_XMLWRITER_H
+
+#define BAD_CAST_CONST (const xmlChar *)
+
+struct xml_writer {
+	xmlTextWriterPtr writer;
+	xmlBufferPtr bp;
+	unsigned int indent;
+};
+
+static int
+xml_writer_create(struct xml_writer **pctx) {
+	struct xml_writer *ctx = calloc(1, sizeof(struct xml_writer));
```

Comment:
Nits:
 - move the opening brace for functions, at the start of the next line
 - drop unused function return type - xml_writer_destroy, xml_writer_set_indent, xml_writer_get_final_content_and_length
 - s/return (x)/return x/ - the style varies across the project, the latter being my personal preference

Reply:
Great catches!

- I'll move the braces. That was my mix-up, Thanks!
- Those returns are required for part 2, https://github.com/DHowett/libarchive/compare/feature/xml-abstraction...DHowett:libarchive:feature/xmllite (the Windows APIs can fail here!)
- Ah, in this case I went with the prevailing style for this _file_ (nit: I forgot to use `()` for a couple bits). Of the 203 `return`s in the document as of 097ecfb8, 100% of them are `()`-style. I'm intending to standardize on that one :smile: even though it's also not my personal style.

Reply:
> Those returns are required for part 2 ...

In that case, they should be introduced as needed. ATM it's dead code that static analyzers and people will flag up.